cdap-proto/src/test/java/co/cask/cdap/proto/id/EntityIdTest.java

```diff
@@ -113,10 +125,36 @@ public void testInvalidId() {
   }
 
   @Test
+  public void testToString() {
+    Assert.assertEquals("namespace:foo", Ids.namespace("foo").toString());
+    Assert.assertEquals("artifact:foo.art.1.2.3", Ids.namespace("foo").artifact("art", "1.2.3").toString());
+    Assert.assertEquals("dataset:foo.zoo", Ids.namespace("foo").dataset("zoo").toString());
+    Assert.assertEquals("dataset_module:foo.moo", Ids.namespace("foo").datasetModule("moo").toString());
+    Assert.assertEquals("dataset_type:foo.typ", Ids.namespace("foo").datasetType("typ").toString());
+    Assert.assertEquals("stream:foo.sdf", Ids.namespace("foo").stream("sdf").toString());
+    Assert.assertEquals("application:foo.app", Ids.namespace("foo").app("app").toString());
+    Assert.assertEquals("program:foo.app.flow.flo", Ids.namespace("foo").app("app").flow("flo").toString());
+    Assert.assertEquals("program_run:foo.app.flow.flo.run1",
+                        Ids.namespace("foo").app("app").flow("flo").run("run1").toString());
+    Assert.assertEquals("flowlet:foo.app.flo.flol",
+                        Ids.namespace("foo").app("app").flow("flo").flowlet("flol").toString());
+    Assert.assertEquals("flowlet_queue:foo.app.flo.flol.q",
+                        Ids.namespace("foo").app("app").flow("flo").flowlet("flol").queue("q").toString());
+  }
+
+  @Test
   @Ignore
   public void testPrintToString() {
```

Comment:
I don't want to maintain a comment on `toString()`, and I'd like a place where I can see all the `EntityId.toString()`s together. What do you suggest?